library/src/permutation/device/hiptensor_permutation_scale_instances.hpp

```diff
@@ -0,0 +1,396 @@
+                                                                                      UnaryOperation,
+                                                                                      Scale,
+                                                                                      2>{});
+
+                }
+
+                template <typename ElementwiseOperation,
+                          typename UnaryOperation,
+                          typename Scale>
+                void add_device_permute_scale_2d_f32_instances(
+                    std::vector<std::unique_ptr<DeviceElementwise<ck::Tuple<F32>,
+                                                                  ck::Tuple<F32>,
+                                                                  ElementwiseOperation,
+                                                                  UnaryOperation,
+                                                                  Scale,
+                                                                  2>>>& instances)
+                {
+                    add_device_operation_instances(instances,
+                                                   device_permute_scale_f32_instances<ElementwiseOperation,
+                                                                                      UnaryOperation,
+                                                                                      Scale,
+                                                                                      2>{});
+                }
+
+                template <typename ElementwiseOperation,
+                          typename UnaryOperation,
+                          typename Scale>
+                void add_device_permute_scale_3d_f16_instances(
+                    std::vector<std::unique_ptr<DeviceElementwise<ck::Tuple<F16>,
+                                                                  ck::Tuple<F16>,
+                                                                  ElementwiseOperation,
+                                                                  UnaryOperation,
+                                                                  Scale,
+                                                                  3>>>& instances)
+                {
+                    add_device_operation_instances(instances,
+                                                   device_permute_scale_f16_instances<ElementwiseOperation,
+                                                                                      UnaryOperation,
+                                                                                      Scale,
+                                                                                      3>{});
+                }
+
+                template <typename ElementwiseOperation,
+                          typename UnaryOperation,
+                          typename Scale>
+                void add_device_permute_scale_3d_f32_instances(
```

Comment:
This is similar to CK's implementation of instances. This split in the early stage based on dims and types allows us to have seperate instances for each category, in case they add more in future for any specific category, it will be easier to add or remove. Thats why the same structure is maintained.